### PR TITLE
[#786] Handle denied state of a plan request (no conversion hosts configured)

### LINKED ISSUE #739: [v2v] Fail migration if no conversion host configured
BZ link:  https://bugzilla.redhat.com/show_bug.cgi?id=1640816

Issue:
When initiating a migration plan, if there are no conversion hosts available the migration plan spins until the 40K timeout has been reached.

Resolution:
If there is no conversion host available fail the migration and provide a reason why in the UI.

### LINKED ISSUE #786: [In Progress Cards] Migration should fail when denied due to lack of conversion host
If a migration is kicked off where there is no conversion host configured in the target environment, the request will be denied.

We need to investigate what happens in this case and make sure that the In Progress Card goes away and the proper notification(s) are shown to the user

https://bugzilla.redhat.com/show_bug.cgi?id=1640816

Duplicate of #739

### PR DESCRIPTION
**Pull request by mturley**
Fixes #786.
Fixes #739.

https://bugzilla.redhat.com/show_bug.cgi?id=1640816

- [x] Display an error state in the MigrationsInProgressCard when the request state is `denied`
- [x] Add a Cancel Migration button in the card footer
- [x] Solve weird CSS problems with the card footer and card height matching (@michaelkro I'm giving up on this for today, could use some help if you have time soon)
- [x] Test with real denied plan requests
- [x] Handle click on the Cancel Migration button
- [x] Remove `test-denied-state` condition
- [ ] Update the `DOCS_URL_CONFIGURE_CONVERSION_HOSTS` URL, which is now a 404

# Screens

A denied plan in the Plans In Progress view:

screenshot 2018-11-19 17 34 06

A denied plan in the Plans Complete view:

screenshot 2018-11-19 17 34 13

A denied plan in the Plan Request Details view:

screenshot 2018-11-19 17 50 06

# Recording

ozj4gxxmvz

### CODE REVIEW

**Comment by mturley** [comment]
The card currently looks like this, and all attempts I've made so far at fixing that button positioning and footer sizing have caused the height-matching to squash the card height down to a point where the footer goes back to not being tall enough. 😡 

screenshot 2018-11-16 16 41 14

**Comment by vconzola** [comment]
This will also fix #739.

File: app/javascript/react/screens/App/Overview/OverviewSelectors.js [hunk 1]
```diff
@@ -19,7 +23,11 @@ export const finishedTransformationPlansFilter = transformationPlans =>
   transformationPlans.filter(transformationPlan => {
     if (transformationPlan.miq_requests.length > 0) {
       const mostRecentRequest = getMostRecentRequest(transformationPlan.miq_requests);
-      return mostRecentRequest.request_state === 'finished' || mostRecentRequest.request_state === 'failed';
+      return (
+        (mostRecentRequest.request_state === 'finished' && mostRecentRequest.approval_state !== 'denied') ||
+        mostRecentRequest.request_state === 'failed' ||
+        (mostRecentRequest.approval_state === 'denied' && mostRecentRequest.options.denial_acknowledged)
```
Comment (michaelkro) [inline app/javascript/react/screens/App/Overview/OverviewSelectors.js]: Methinks we can remove the `approval_state` check here

**Comment by mzazrivec** [comment]
@simaishi Could this be cherry-picked into hammer please? Thank you.

**Comment by mzazrivec** [comment]
@simaishi Could this be cherry-picked into hammer please? Thank you.

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 9b78d0947e0f096e0a6ee8f8d160acef10ebf089
Author: Michael Ro <mikerodev@gmail.com>
Date:   Wed Nov 21 08:19:43 2018 -0800

    Merge pull request #798 from mturley/786-plan-denied-state
    
    [#786] Handle denied state of a plan request (no conversion hosts configured)
    
    (cherry picked from commit e8e89d19159472f48d4d3c1fbd37545142ed541a)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1640816
```

**Pull request merged**